ScienceJournal/Chart/ChartController.swift

```diff
@@ -481,9 +481,17 @@ class ChartController: NSObject, ChartViewDelegate, UIScrollViewDelegate {
     let increment = axisLabels[1] - axisLabels[0]
     guard increment > 0 else { return 0 }
 
-    let startIndex = Int(floor((minShown - axisLabels[0]) / increment + 1))
-    let endIndex = Int(ceil((maxShown - lastAxisLabel) / increment)) + axisLabels.count - 1
-    return endIndex - startIndex
+
+    let startIndex = Int(exactly: floor((minShown - axisLabels[0]) / Double(increment + 1)))
+    let endIndexCeil = ceil((maxShown - lastAxisLabel) / increment)
+    let endIndex = Int(exactly: endIndexCeil + Double(axisLabels.count - 1))
+
+    if let startIndex = startIndex, let endIndex = endIndex {
+      // If percentage is 1.0 index can be beyond count so clamp it.
```

Comment:
This comment looks like it was copied from a different context.

Reply:
Whoopsies.